lib/jasmine/jasmine.ts

```diff
@@ -17,10 +12,14 @@ const _global = typeof window == 'undefined' ? global : window;
 // it. We override the `clearStack` method which forces jasmine to always
 // drain the stack before next test gets executed.
 (<any>jasmine).QueueRunner = (function (SuperQueueRunner) {
+  const originalZone = Zone.current;
   // Subclass the `QueueRunner` and override the `clearStack` method.
 
   function alwaysClearStack(fn) {
-    _global[SET_TIMEOUT](fn, 0);
+    const zone: Zone = Zone.current.getZone('JasmineClearStackZone')
+        || Zone.current.getZone('ProxyZoneSpec')
+        || originalZone;
+    zone.scheduleMicroTask('jasmineCleanStack', fn);
```

Comment:
Good observation, but it will not happen. The reason is that is is the top most frame which  does the microtask execution, and so it will never nest. This method will always return and only later execute the microtask.